src/main/java/bio/terra/cloudfunctions/streaming/GcsBQ.java

```diff
@@ -0,0 +1,232 @@
+package bio.terra.cloudfunctions.streaming;
+
+import bio.terra.cloudevents.GCSEvent;
+import com.google.cloud.ReadChannel;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FormatOptions;
+import com.google.cloud.bigquery.JobInfo;
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.bigquery.TableDataWriteChannel;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.WriteChannelConfiguration;
+import com.google.cloud.bigquery.testing.RemoteBigQueryHelper;
+import com.google.cloud.functions.BackgroundFunction;
+import com.google.cloud.functions.Context;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageException;
+import com.google.cloud.storage.StorageOptions;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.logging.Logger;
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.ArchiveStreamFactory;
+import org.apache.commons.compress.compressors.CompressorInputStream;
+import org.apache.commons.compress.compressors.CompressorStreamFactory;
+
+public class GcsBQ implements BackgroundFunction<GCSEvent> {
```

Comment:
Assuming you're ok with my suggestions for `GoogleCloudStorageEventHarness` I'd be fine with moving this business logic to that paradigm.

Reply:
`GcsBQ` has now been replaced by `TestRunnerBackgroundFunction`.